tests/unittests/GraphTest.cpp

```diff
@@ -1324,6 +1324,58 @@ TEST(Graph, cloneContextRuns) {
   EXPECT_FALSE(saveBacking1->isEqual(*saveBacking2));
 }
 
+TEST(Graph, ContextDoubleAllocate) {
```

Comment:
Graph test is pretty giant now, maybe extract context specific tests into a separate file.